ext/pcntl/pcntl.c

```diff
@@ -1453,6 +1457,59 @@ PHP_FUNCTION(pcntl_unshare)
 /* }}} */
 #endif
 
+#ifdef HAVE_CAP_ENTER
+/* {{{ proto bool pcntl_cap_enter()
+   current process enter in capability mode */
+PHP_FUNCTION(pcntl_cap_enter)
+{
+	int ret;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	ret = cap_enter();
+	if (ret == -1) {
+		// ENOSYS is the only reason
+		// Captured at configure time but in case the support was removed afterwards.
+		PCNTL_G(last_error) = errno;
+		php_error_docref(NULL, E_WARNING, "Error %d: capability mode unsupported", errno);
+		RETURN_FALSE;
+	}
+	RETURN_TRUE;
+}
+/* }}} */
+
+/* {{{ proto bool pcntl_cap_getmode()
+   checks current process capability mode */
+PHP_FUNCTION(pcntl_cap_getmode)
```

Comment:
I realize that this is the name of the C API, but given how it just returns a boolean, maybe we can make the name more meaningful? I'm thinking `pcntl_cap_enabled()`.

Reply:
sure why not